### PR TITLE
Fix crash in inner core example

### PR DESCRIPTION
**Pull request by tjhei**
The conversion to only assembling Stokes DoFs was not done for this
example (but the copy in tests/ is correct).

### CODE REVIEW

File: cookbooks/inner_core_convection/inner_core_assembly.cc [hunk 1]
```diff
@@ -54,30 +54,44 @@ namespace aspect
                                         internal::Assembly::CopyData::StokesSystem<dim>      &data) const
       {
         const Introspection<dim> &introspection = this->introspection();
+        const FiniteElement<dim> &fe = this->get_fe();
+        const unsigned int stokes_dofs_per_cell = data.local_dof_indices.size();
+        const unsigned int n_q_points           = scratch.face_finite_element_values.n_quadrature_points;
```
Comment (bangerth) [inline cookbooks/inner_core_convection/inner_core_assembly.cc]: Can you indent line 57 so the equal signs line up?

**Comment by bangerth** [comment]
I have to admit that I don't know the details of why your changes are necessary, but just for completeness: the results you get with the patch are the same as the ones shown in the manual?

**Comment by tjhei** [comment]
> I have to admit that I don't know the details of why your changes are necessary, but just for completeness:

Because data.local_matrix is now smaller (only contains Stokes DoFs) and we get an out of bounds error when writing zeros otherwise.

> he results you get with the patch are the same as the ones shown in the manual?

I have not tried that. It just doesn't crash in timestep 0 anymore.

**Comment by bangerth** [comment]
> Because data.local_matrix is now smaller (only contains Stokes DoFs) and
> we get an out of bounds error when writing zeros otherwise.

Oh yes, I remember now. Thanks for the refresher.

I think this is strictly progress. It would of course be nice to verify that it actually works. @jdannberg , what do you think?

**Comment by gassmoeller** [comment]
This looks correct. The changes look visually the same as in tests/inner_core.cc, and that one was changed by me without affecting the test results.

**Comment by jdannberg** [comment]
I think what happened is that at some point we made changes to the code and updated the test (but not the cookbook). As these changes are exactly the same ones as in the test, and the inner core test case is still working, this should be fine (and we should also still get mostly the same results).

**Comment by bangerth** [comment]
OK, great. OK to merge then once the tester is happy.

**Comment by tjhei** [comment]
> The changes look visually the same as in tests/inner_core.cc

No coincidence: I first fixed the code manually, then realized "hey, we need a test for that... Wait a minute, there is already a test for that! Why does the test work? Oh, let's copy and paste the assembly from there." ;-)

**Pull request merged**